crates/sys/src/furi.rs

```diff
@@ -60,11 +60,11 @@ impl Status {
     }
 
     /// Returns `Err(Status)` if [`Status`] is an error, otherwise `Ok(or_else(Status))`.
-    pub fn err_or_else<T>(self, or_else: impl Fn(Self) -> T) -> Result<T, Self> {
+    pub fn err_or_else<T>(self, or_else: impl Fn(i32) -> T) -> Result<T, Self> {
```

Comment:
The trouble with `NonZeroI32` is that a zero status is typically considered a success value, thus should result in an `Ok(...)` rather than a `Err(Status(0))`.

I agree that `err_or_else` is a bit of an awkward name. It was intended to mirror [`Option::ok_or_else`](https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or_else), but I'm still not a huge fan of it. Perhaps something like `map_or_err` would be a better fit..

Reply:
Ah. I see that I defined `is_err` as any non-status `OK` value, which is technically correct but confusing if you can construct  `Status` from arbitrary `i32` values.

Hmmm... So there's a couple of ways that we could treat `Status`:
1. `Status` should only hold status values, thus you should never be able to construct one with a positive value
2. `Status` can be overloaded with a positive return value

I think my preferred approach would be to have a `furi::Error` type that is an enum of strictly error values, rather than trying to use `Status` like a poor substitute for `Result`.